extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanServiceBindingConverter.java

```diff
@@ -0,0 +1,102 @@
+package io.quarkus.infinispan.client.runtime;
+
+import static java.lang.String.format;
+import static org.apache.sshd.common.util.GenericUtils.isBlank;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBinding;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConfigSource;
+import io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter;
+
+/**
+ * ServiceBindingConverter for Infinispan to support SBO (Service Binding Operator) in Quarkus.
+ *
+ * <br>
+ * <br>
+ * Following individual properties are supported to make the connection:
+ * <ul>
+ * <li>username</li>
+ * <li>password</li>
+ * <li>useauth</li>
+ * <li>server (<i>server information can be provided in this property separated by : sign</i>, e.g. localhost:11222</li>
+ * </ul>
+ * <i>Other than server all other properties are optional</i>
+ *
+ * <br>
+ * <br>
+ * The Quarkus properties set by this class are:
+ * <ul>
+ * <li>quarkus.infinispan-client.server-list</li>
+ * <li>quarkus.infinispan-client.use-auth (<i>if useauth is provided</i>)</li>
+ * <li>quarkus.infinispan-client.auth-username (<i>if username is provided</i>)</li>
+ * <li>quarkus.infinispan-client.auth-password (<i>if password is provided</i>)</li>
+ * </ul>
+ */
+public class InfinispanServiceBindingConverter implements ServiceBindingConverter {
+
+    private static final Logger LOGGER = Logger.getLogger(InfinispanServiceBindingConverter.class);
+    private static final String BINDING_TYPE = "infinispan";
+    public static final String BINDING_CONFIG_SOURCE_NAME = BINDING_TYPE + "-k8s-service-binding-source";
+
+    // Connection properties
+    public static final String INFINISPAN_SERVER_LIST = "serverlist";
+    public static final String INFINISPAN_USE_AUTH = "useauth";
```

Comment:
How about just `auth`  and the value specified in this file could then be used to determine the auth mechanism to use? If the value is `false`, then we set `use-auth=false`.

Reply:
the evolution should go into the extension configuration too so both are equivalent. something to think about

Reply:
The intent of ServiceBinding is not to provide a 1-1 mapping between files and the existing properties so I don't this needs to be the same tbh. A ServiceBinding secret containing the file mappings can either be explicitly created by a user, or more likely, will be provisioned by a ServiceBinding provider (Infinispan operator). So only providing mappings for a small number of bindings is fine as long as the server connection can be established.